test_libs/pyspec/eth2spec/utils/ssz/ssz_impl.py

```diff
@@ -14,68 +10,47 @@
 BYTES_PER_LENGTH_OFFSET = 4
 
 
-def is_basic_type(typ):
-    return is_uint_type(typ) or is_bool_type(typ)
-
-
-def serialize_basic(value, typ):
-    if is_uint_type(typ):
-        return value.to_bytes(uint_byte_size(typ), 'little')
-    elif is_bool_type(typ):
+def serialize_basic(value: SSZValue):
+    if isinstance(value, uint):
+        return value.to_bytes(value.type().byte_len, 'little')
+    elif isinstance(value, Bit):
         if value:
             return b'\x01'
         else:
             return b'\x00'
     else:
-        raise Exception("Type not supported: {}".format(typ))
+        raise Exception(f"Type not supported: {type(value)}")
 
 
-def deserialize_basic(value, typ):
-    if is_uint_type(typ):
+def deserialize_basic(value, typ: BasicType):
+    if issubclass(typ, uint):
```

Comment:
why use `issubclass` here but `isinstance` in `serialize_basic`?

Reply:
Values are instances of classes. types are subclasses of classes. Types are instances of type-subclasses (meta classes). Value here is a a python value, hence the type is added as an argument here, to describe what to handle the value like.